File: web/src/dashboard/status.ts
const COLORS: Record<string, string> = {
  queued: "gray", running: "blue", blocked: "red", paused: "amber",
  completed: "green", failed: "red", cancelled: "gray",
};

export function statusColor(status: string): string {
  return COLORS[status] ?? "gray";
}

const TERMINAL = new Set(["completed", "failed", "cancelled"]);
export function isTerminalStatus(status: string): boolean {
  return TERMINAL.has(status);
}

File: ios/Autoloop/Domain/Status.swift
import Foundation

/// Semantic status colors (mirrors the web's color classes; mapped to real colors in Theme).
enum StatusColor { case gray, blue, red, amber, green }

private let statusColors: [String: StatusColor] = [
    "queued": .gray, "running": .blue, "blocked": .red, "paused": .amber,
    "completed": .green, "failed": .red, "cancelled": .gray,
]

func statusColor(_ status: String) -> StatusColor { statusColors[status] ?? .gray }

private let terminalStatuses: Set<String> = ["completed", "failed", "cancelled"]
func isTerminalStatus(_ status: String) -> Bool { terminalStatuses.contains(status) }
